### PR TITLE
[ffi] Support alignment on objects

### LINKED ISSUE #736: Add support for defining alignment of FFI structs
This would mean adding `#[repr(align(XXX))]` to the struct definitions, e.g.

```rust
#[repr(C)]
#[repr(align(16))]
struct Foo {
    x: u8,
}
```

Currently this would have to be done via configuration, but ideally there would be a gobject-introspection annotation for that.

@EPashkin Can we do per-type configuration in -sys mode right now?

### PR DESCRIPTION
**Pull request by sfanxiang**
Closes #736.

### CODE REVIEW

File: src/codegen/sys/lib_.rs [hunk 1]
```diff
@@ -333,8 +333,11 @@ fn generate_unions(w: &mut Write, env: &Env, unions: &[&Union]) -> Result<()> {
         if !env.type_status_sys(&full_name).need_generate() {
             continue;
         }
+        let config = env.config.objects.get(&full_name);
+        let align = config.and_then(|c| c.align);
```
Comment (EPashkin) [inline src/codegen/sys/lib_.rs]: Can you move `config` acquisition before status check and use it instead `env.type_status_sys`,
maybe with some new function in `Env`
```
pub fn unwrap_status_sys(status: Option<GStatus>) -> GStatus {
    status.unwrap_or(GStatus::Generate)
}
```

File: src/config/gobjects.rs [hunk 2]
```diff
@@ -287,6 +290,17 @@ fn parse_object(
                 .collect()
         })
         .unwrap_or_else(|_| Vec::new());
+    let align = toml_object
+        .lookup("align")
+        .and_then(|v| v.as_integer())
+        .and_then(|v| {
+            if v.count_ones() != 1 || v > i64::from(u32::max_value()) || v < 0 {
+                warn!("align configuration must be a power of two of type u32, found {}", v);
```
Comment (EPashkin) [inline src/config/gobjects.rs]: Please use "\`align\` configuration ..." like in https://github.com/gtk-rs/gir/blob/master/src/config/error.rs#L62

**Comment by EPashkin** [comment]
Thanks.

File: src/codegen/sys/lib_.rs (hunk 1 above)
Comment (EPashkin) [reply src/codegen/sys/lib_.rs]: .. or in global namespace with code near `GStatus` definition

Comment (EPashkin) [reply src/codegen/sys/lib_.rs]: Sorry, forgot about `.map(|o| o.status)`
```
pub fn unwrap_status_sys(object: Option<GObject>) -> GStatus {
    if let Some(object) = object {
       object.status
    } else {
        GStatus::Generate
    }
}
```

Comment (sfanxiang) [reply src/codegen/sys/lib_.rs]: Other code in this file does it in a way that makes it easier to spot what's going on without increasing LOC:

```rust
if let Some(false) = config.map(|c| c.status.need_generate()) {
    continue;
}
```

I actually prefer this one. Let me know if you still want `unwrap_status_sys()`.

Comment (EPashkin) [reply src/codegen/sys/lib_.rs]: This is better solution and `Env::type_status_sys` maybe unneeded now. Thanks.

Comment (EPashkin) [reply src/codegen/sys/lib_.rs]: It was almost year ago :wink:
https://github.com/gtk-rs/gir/commit/032ec3fcc3cea62166c7a1cd941ff38946a18115#diff-0c37b0c87c3c91119e579feee853b5d0R187

**Comment by EPashkin** [comment]
Thanks

**Pull request merged**